src/interfaces/IOptionSettlementEngine.sol

```diff
@@ -4,54 +4,94 @@ pragma solidity ^0.8.0;
 import "./IERC1155Metadata.sol";
 
 // TODO(The engine is IERC1155Metadata, but the solmate impl is not compatible with interface, re-implement)
-// @title A settlement engine for options
-// @author 0xAlcibiades
+/// @title A settlement engine for options
+/// @author 0xAlcibiades
 interface IOptionSettlementEngine {
-    // The requested token is not found.
-    // @param token token requested.
+    /**
+     * @notice The requested token is not found.
+     * @param token token requested.
+     */
     error TokenNotFound(uint256 token);
 
-    // The caller doesn't have permission to access that function.
+    /**
+     * @notice The caller doesn't have permission to access that function.
+     * @param accessor The requesting address.
+     * @param permissioned The address which has the requisite permissions.
+     */
     error AccessControlViolation(address accessor, address permissioned);
 
-    // This options chain already exists and thus cannot be created.
+    /**
+     * @notice This options chain already exists and thus cannot be created.
+     * @param hash The hash of the options chain.
+     */
     error OptionsChainExists(bytes32 hash);
 
-    // The expiry timestamp is less than 24 hours from now.
+    /// @notice The expiry timestamp is less than 24 hours from now.
     error ExpiryTooSoon();
 
-    // The option exercise window is less than 24 hours long.
+    /// @notice The option exercise window is less than 24 hours long.
     error ExerciseWindowTooShort();
 
-    // The assets specified are invalid or duplicate.
+    /**
+     * @notice The assets specified are invalid or duplicate.
+     * @param asset1 Supplied asset.
+     * @param asset2 Supplied asset.
+     */
     error InvalidAssets(address asset1, address asset2);
 
-    // The token specified is not an option.
+    /**
+     * @notice The token specified is not an option.
+     * @param token The supplied token.
+     */
     error InvalidOption(uint256 token);
 
-    // The token specified is not a claim.
+    /**
+     * @notice The token specified is not a claim.
+     * @param token The supplied token.
+     */
     error InvalidClaim(uint256 token);
 
-    // The optionId specified expired at expiry.
+    /**
+     * @notice The optionId specified expired at expiry.
+     * @param optionId The id of the expired option.
+     * @param expiry The time of expiry of the supplied option Id.
+     */
     error ExpiredOption(uint256 optionId, uint40 expiry);
 
-    // This option cannot yet be exercised.
+    /// @notice This option cannot yet be exercised.
     error ExerciseTooEarly();
```

Comment:
I know this is just documentation updates but there should probably be an associated optionId param for this event.

Reply:
Do you want expiry on that as well?